<launch>
  <param name="/use_sim_time" value="true" />
  <!-- 15, 15, 0; // density01seed0_no_ground -->
  <!-- 20, 0, 0.5; // bugtrap10 -->

  <!-- icra0: 27, 12 -->
  <!-- zigzag7: 20, 22 -->
  <!-- bugtrap10: 20, 0 -->
  <!-- perception: -3, -3 -->
  <arg name="map_offset_x_" default="27"/>
  <arg name="map_offset_y_" default="12"/> 
  <arg name="map_offset_z_" default="0.5"/>
  <!-- World 文件路径 -->
  <arg name="world_file" default="/home/bhrqhb/catkin_ws/planner/catkin_ws_TGH_Planner/src/TGH-Planner/uav_simulator/gazebo_simulator/world/icra0.world" />

  <!-- 启动 Gazebo -->
  <node name="gazebo" pkg="gazebo_ros" type="gzserver" args="-s libgazebo_ros_factory.so -world $(arg world_file) -gui false" output="screen"/>
  
  <!-- If you want to open gazebo gui -->
  <node name="gazebo_gui" pkg="gazebo_ros" type="gzclient" args="" output="screen"/>

  <!-- 加载 URDF 模型 -->
  <param name="robot_description" textfile="$(find gazebo_simulator)/urdf/box.urdf" />
  
  <!-- 使用 robot_description 加载模型到 Gazebo -->
  <node name="spawn_model" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model simple_box" output="screen" />

  <!-- 启动自定义节点 -->
  <node name="gazebo_simulator" pkg="gazebo_simulator" type="set_robot_pose" output="screen" >
    <param name="map_offset_x" value="$(arg map_offset_x_)" type="double"/>
    <param name="map_offset_y" value="$(arg map_offset_y_)" type="double"/>
    <param name="map_offset_z" value="$(arg map_offset_z_)" type="double"/>
  </node>
</launch>
